src/core/agent/services/identifierService.ts

```diff
@@ -328,18 +328,115 @@ class IdentifierService extends AgentService {
       (identifier: IdentifierResult) =>
         !storageIdentifiers.find((item) => identifier.prefix === item.id)
     );
-    if (unSyncedData.length) {
-      //sync the storage with the signify data
-      for (const identifier of unSyncedData) {
+    
+    const [unSyncedDataWithGroup, unSyncedDataWithoutGroup] = [
+      unSyncedData.filter((item: HabState) => item.group !== undefined),
+      unSyncedData.filter((item: HabState) => item.group === undefined),
+    ];
+
+    if (unSyncedDataWithoutGroup.length) {
```

Comment:
We don't need these if statements since it'll always be `[]` if empty. Here and below